lib/ukrand/chacha.c

```diff
@@ -116,7 +116,19 @@ static inline __u32 _infvec_val(unsigned int c, const __u32 v[],
 	return v[pos % c];
 }
 
-void uk_swrand_init_r(struct uk_swrand *r, unsigned int seedc,
+__u32 uk_chacha_gen_seed32(void)
+{
+	int ret;
+	__u32 val;
+
+	ret = ukarch_random_seed_u32(&val);
+	if (unlikely(ret))
+		UK_CRASH("Could not generate random seed\n");
```

Comment:
I think it should be the responsibility of the caller to know whether it's worth crashing the system or not on a failed function call :thinking:. I think the main issue here is the mixed semantics: `ukarch_random_seed_u32` seems to return the actual result through the argument and an error code through the actual return value of the functio, while `uk_chacha_gen_seed32` takes no arguments and supposedly returns the actual result through the actual return value of the function.

Therefore, I would maintain the same "signature" for the two functions. As far as I can tell from reading this function, it is supposed to be a wrapper. It does not seem to be doing anything other than call  `ukarch_random_seed_u32` itself (at least at the moment? maybe there is more planned?). Although I do not have as much context as you guys since I did not attend all the meetings. @michpappas WDYT?

Reply:
You are right, I did not notice that `uk_chacha_gen_seed32` was just a wrapper for `ukarch_random_seed_u32`. I removed the wrapper.